main.go

```diff
@@ -56,138 +44,211 @@ func downloadCacheArchive(url string) error {
 	if resp.StatusCode != 200 {
 		responseBytes, err := ioutil.ReadAll(resp.Body)
 		if err != nil {
-			return err
+			return "", err
 		}
 
-		return fmt.Errorf("non success response code: %d, body: %s", resp.StatusCode, string(responseBytes))
+		return "", fmt.Errorf("non success response code: %d, body: %s", resp.StatusCode, string(responseBytes))
 	}
 
-	out, err := os.Create(cacheArchivePath)
+	const cacheArchivePath = "/tmp/cache-archive.tar"
+	f, err := os.Create(cacheArchivePath)
 	if err != nil {
-		return fmt.Errorf("failed to open the local cache file for write: %s", err)
+		return "", fmt.Errorf("failed to open the local cache file for write: %s", err)
 	}
 
-	defer func() {
-		if err := out.Close(); err != nil {
-			log.Printf(" [!] Failed to close Archive download file: %+v", err)
-		}
-	}()
-
-	_, err = io.Copy(out, resp.Body)
+	_, err = io.Copy(f, resp.Body)
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	return nil
+	return cacheArchivePath, nil
 }
 
-func uncompressArchive() error {
-	cmd := command.New("tar", "-xPf", cacheArchivePath)
-	return cmd.Run()
-}
-
-func downloadAndExtractCacheArchive(url string) error {
+// performRequest performs an http request and returns the response's body, if the status code is 200.
+func performRequest(url string) (io.ReadCloser, error) {
 	resp, err := http.Get(url)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	if resp.StatusCode != 200 {
+		defer func() {
+			if err := resp.Body.Close(); err != nil {
+				log.Warnf("Failed to close response body, error: %s", err)
```

Comment:
Use the same syntax please for closing errors:
`Printf([!] Failed to close Archive download response body: %s", err))`

vs

`Warnf("Failed to close response body, error: %s", err)`